### PR TITLE
Replace HoneycombSdk with OpenTelemetryConfiguration.Builder

### LINKED ISSUE #130: Make Honey SDK a pure configurer
The (currently known as) Honeycomb SDK should return an instance of OpenTelemetry class, instead of our own "SDK" class. Similarly to the .NET distro, we should simply configure the OTEL instances, and allow users to interact with OTEL interfaces from then on.

- this is a breaking change
- do we keep the module/artifact name? ("SDK") - changing the name might make it less confusing to users

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
## Which problem is this PR solving?
Removes the need to create and maintenance a custom SDK wrapper. Instead the new `OpenTelemetryConfiguration` builder configures and returns a base vanilla `OpenTelemetrySdk` instance.

This means an application that configures the SDK using this package doesn't need to be aware of a custom SDK wrapper and can use the base OpenTelemetrySdk as normal.

- Closes #130 

## Short description of the changes
- Updates opentelemetry-java and opentelemetry-java-agent dependencies to 1.5.0 and 1.5.3
- Rename `HoneycombSdk` to `OpenTelemetryConfiguration`
- Don't wrap the `OpenTelemetrySdk` with a custom class, instead configure and return an instance of the `OpenTelemetrySdk` when calling `build`
- Adds a simple Makefile with actions for build, test and publish_local
- Update .gitignore to not track Eclipse based files (.settings, .classpath, etc) or sub-component (agent, sdk, common, etc) bin folders

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,8 +10,8 @@ This is Honeycomb's distribution of OpenTelemetry for Java.
 It makes getting started with OpenTelemetry and Honeycomb easier!
 
 Latest release built with:
-- [OpenTelemetry](https://github.com/open-telemetry/opentelemetry-java/releases/tag/v1.2.0) version 1.2.0
-- [OpenTelemetry Java Agent](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v1.2.0) version 1.2.0
+- [OpenTelemetry](https://github.com/open-telemetry/opentelemetry-java/releases/tag/v1.5.0) version 1.5.0
```
Comment (vreynolds) [inline README.md:13]: I think we should wait to update these until it's time for a release

File: sdk/src/main/java/io/honeycomb/opentelemetry/OpenTelemetryConfiguration.java [hunk 2]
```diff
@@ -218,24 +194,24 @@ public AttributesBuilder addResourceAttribute(String key, String... value) {
...
-         * Returns a new {@link HoneycombSdk} built with the configuration of this {@link
+         * Returns a new {@link OpenTelemetry} built with the configuration of this {@link
          * Builder} and registers it as the global {@link
          * io.opentelemetry.api.OpenTelemetry}. An exception will be thrown if this method is attempted to
          * be called multiple times in the lifecycle of an application - ensure you have only one SDK for
          * use as the global instance. If you need to configure multiple SDKs for tests, use {@link
          * GlobalOpenTelemetry#resetForTest()} between them.
          *
-         * @return {@link HoneycombSdk} instance
+         * @return {@link OpenTelemetryConfiguration} instance
```
Comment (vreynolds) [inline sdk/src/main/java/io/honeycomb/opentelemetry/OpenTelemetryConfiguration.java:204]:
```suggestion
         * @return {@link OpenTelemetry} instance
```

File: sdk/src/main/java/io/honeycomb/opentelemetry/OpenTelemetryConfiguration.java [hunk 3]
```diff
@@ -245,13 +221,13 @@ public HoneycombSdk buildAndRegisterGlobal() {
          * {@link SdkTracerProvider} with a {@link BatchSpanProcessor} that has an
          * {@link OtlpGrpcSpanExporter} configured.</p>
          *
-         * @return {@link HoneycombSdk} instance
+         * @return {@link OpenTelemetryConfiguration} instance
```
Comment (vreynolds) [inline sdk/src/main/java/io/honeycomb/opentelemetry/OpenTelemetryConfiguration.java:224]:
```suggestion
         * @return {@link OpenTelemetry} instance
```

**Comment by vreynolds** [comment]
This would be a major bump, yes?

File: README.md (hunk 1 above)
Comment (MikeGoldsmith) [reply README.md:13]: I can pull these into a separate PR to merge when ready. They don't have a material impact on the refactor.

**Comment by MikeGoldsmith** [comment]
> This would be a major bump, yes?

I think a minor is okay here because still in beta, eg we're are release 0.4.0. Things break in beta 😁 

This could be released early to get feedback as a 0.5.0, or be rolled up into the 1.0.0 GA.

File: README.md (hunk 1 above)
Comment (MikeGoldsmith) [reply README.md]: Moved updating OTel deps to #134

**Review by vreynolds — APPROVED** [review]

**Pull request merged**